provider/azure/internal/errorutils/errors.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package errorutils
+
+import (
+	"github.com/juju/errors"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+// ServiceError returns the *azure.ServiceError underlying the
+// supplied error, if any, and a bool indicating whether one
+// was found.
+func ServiceError(err error) (*azure.ServiceError, bool) {
+	err = errors.Cause(err)
+	if d, ok := err.(autorest.DetailedError); ok {
+		err = d.Original
+	}
+	if err, ok := err.(*azure.RequestError); ok {
```

Comment:
Maybe call the new `err` here something else to make this clearer? `reqErr` or just `r` (as per `d` above).